tracer/src/Datadog.Tracer.Native/fault_tolerant_method_duplicator.cpp

```diff
@@ -0,0 +1,348 @@
+#include "fault_tolerant_method_duplicator.h"
+
+#include "cor_profiler.h"
+#include "dd_profiler_constants.h"
+#include "fault_tolerant_envionrment_variables_util.h"
+#include "fault_tolerant_tracker.h"
+#include "logger.h"
+
+fault_tolerant::FaultTolerantMethodDuplicator::FaultTolerantMethodDuplicator(CorProfiler* corProfiler,
+    std::shared_ptr<trace::RejitHandler> rejit_handler, std::shared_ptr<trace::RejitWorkOffloader> work_offloader):
+    m_corProfiler(corProfiler),
+    m_rejit_handler(std::move(rejit_handler)),
+    m_work_offloader(std::move(work_offloader)),
+    is_fault_tolerant_instrumentation_enabled(IsFaultTolerantInstrumentationEnabled())
+{
+}
+
+void fault_tolerant::FaultTolerantMethodDuplicator::Duplicate(const ModuleID moduleId, const trace::ModuleInfo& moduleInfo,
```

Comment:
I don't think this is a concern for this PR, but do we need to worry about calling this multiple times if we later store this object in the `TracerRejitPreprocessor`?

Reply:
I don't quite understand the concern you raised🤔, this part has nothing to do with the preprocessors, and the `FaultTolerantDuplicator` itself does not hold any data (it's merely a service class).
Ultimately, `Duplicate` will be called only once per `moduleId` & `methodId` combination (as the call path is coming from `ModuleLoadFinished`) but there is nothing wrong with calling `Duplicate` multiple times on the same method - nothing wrong will happen, the `DefineMethod` API will return that the method already exist.
Does that answer your question/concern?

Reply:
> but there is nothing wrong with calling Duplicate multiple times on the same method - nothing wrong will happen, the DefineMethod API will return that the method already exist.

Yes this was the main concern I had, thanks for going into detail 🙏🏼